### PR TITLE
Consolidate SQRT_ISWAP gate defs

### LINKED ISSUE #4223: Consolidate SQRT_ISWAP gate definitions
**Is your feature request related to a use case or problem? Please describe.**

Recently, #4213 added (will add) the SQRT_ISWAP=ISWAP**0.5 gate to the core set of gates.  Several places throughout Cirq define it and one place even defines it as [ISWAP**-0.5](https://github.com/quantumlib/Cirq/blob/2f8326c5f71006ff17c9353e30d35cd9e11aca2a/cirq-google/cirq_google/calibration/xeb_wrapper_test.py) (see https://github.com/quantumlib/Cirq/search?q=SQRT_ISWAP for the full list) which could lead to inconsistencies or bugs.

**Describe the solution you'd like**
These duplicate definitions `SQRT_ISWAP = ISWAP**0.5` should be removed and all uses of the gate should be updated to `ops.SQRT_ISWAP` or `cirq.SQRT_ISWAP` as appropriate.  https://github.com/quantumlib/Cirq/search?q=SQRT_ISWAP lists all places that need to be updated.

**What is the urgency from your perspective for this issue? Is it blocking important work?**

P2 - we should do it in the next couple of quarters

### PR DESCRIPTION
**Pull request by ShrillShrestha**
Resolves #4223 
@cduck, can you please review it? I think I might need so suggestions on changes.

### CODE REVIEW

**Review by tanujkhattar — CHANGES_REQUESTED** [review]
@ShrillShrestha  Please fix the failing tests and then mark the PR as ready for review. For more details on how to run these tests locally, see [development.md#continuous-integration-and-local-testing](https://github.com/quantumlib/Cirq/blob/master/docs/dev/development.md#continuous-integration-and-local-testing)

**Comment by tanujkhattar** [comment]
Note that your PR depends on a change that has not yet been submitted, i.e. adding the `cirq.SQRT_ISWAP` to the list of common gates available: https://github.com/quantumlib/Cirq/pull/4213/files#diff-773f7fb395203abbbc4f80a5aab18370db2537c7cd626569a65602d12f7a4afcR340

You can either wait for the above PR to get submitted, or update your PR to include change of introducing the `cirq.SQRT_ISWAP` gate and then replace the existing versions to use it. 

If you choose to wait, I'd recommend to either mark your PR as a draft or rebase it on top of the unsubmitted PR so that the tests don't fail.

**Comment by ShrillShrestha** [comment]
@tanujkhattar, thanks for the comment. Since I am kind of new to contributing, would please tell me how I can rebase my changes on top of an unsubmitted PR? Or you can direct me to some resource where I can learn that. Else, I can just convert it into a draft for now.

**Comment by tanujkhattar** [comment]
1. [Checkout the unsubmitted PR as a local branch](https://stackoverflow.com/questions/27567846/how-can-i-check-out-a-github-pull-request-with-git) 
2. [Rebase your branch on top of it](https://stackoverflow.com/questions/40009820/git-rebase-one-branch-on-top-of-another-branch/40009875) 

Though I think it'll be easier if you either copy the original change to your PR or mark it as draft and wait for the dependent PR to get submitted. To test your changes locally for the meantime, you can follow the above steps.

**Comment by ShrillShrestha** [comment]
@tanujkhattar, Thanks. I will make it as a draft for now, and try to test it locally.

**Comment by tanujkhattar** [comment]
@ShrillShrestha The blocking PR has been merged. You can rebase this PR now and fix the tests so that it's ready for review.

**Comment by ShrillShrestha** [comment]
Hey @tanujkhattar, thanks for notifying. I have a quick question: 
I am getting this error (ImportPathMismatchError):
Error
 when I try to run `pytest .`  from the root folder.
 Can you help me on resolving this and running the tests locally?

**Comment by tanujkhattar** [comment]
@ShrillShrestha Did you question get resolved based on the discussions on gitter? Please fix the tests and let me know once it's ready for review.

**Comment by ShrillShrestha** [comment]
@tanujkhattar, I was able to run the tests and all the test are passing locally (Ubuntu 20.04, python 3.7) except the ones that are failing in the master branch itself. Can you please review it and let me know if I need to make any further changes? 

Also, thank you for pointing out Gitter to me. The discussion really helped me to get some insights.

**Comment by ShrillShrestha** [comment]
@cduck, @tanujkhattar, it seems the formatting for the notebook is not correct in my updates. I am not sure how notebook formatting works. Can you give some idea on how I can apply proper formatting for the notebooks?

**Comment by tanujkhattar** [comment]
You should run `./check/nbformat --apply` from the repo root to run an automated notebook formatter.

**Comment by ShrillShrestha** [comment]
Thanks @tanujkhattar, I ran the scripts. The formatting should be good now.

**Comment by ShrillShrestha** [comment]
The `SQRT_ISWAP` attribute doesn't seem to be exposed from `cirq` in notebooks. I tried looking over it, but couldn't find a way to solve it. Can you review the problem and help me solve it?

**Comment by tanujkhattar** [comment]
Note that the notebooks do a `pip install cirq` at the top and are supposed to run independently outside of the repo code. Since the change that you are making is not yet available in the `pip` package, the notebook tests are failing. Please see the error message that you are getting as follows:

> E           If this is a new failure in this notebook due to a new change, that is only available in master for now, consider adding `pip install --pre cirq` instead of `pip install cirq` to this notebook, and exclude it from dev_tools/notebooks/isolated_notebook_test.py.

In order to fix this, do the following:
- Modify `pip install cirq` to `pip install --pre cirq` in the changed notebooks to ensure that they work with the latest master branch after the PR is merged. 
- Add the path of the modified notebooks to `NOTEBOOKS_DEPENDING_ON_UNRELEASED_FEATURES` in `dev_tools/notebooks/isolated_notebook_test.py.` to exclude them from the `isolated_notebook_tests`. 

See https://github.com/quantumlib/Cirq/blob/master/docs/dev/notebooks.md#lifecycle for more details.

**Comment by tanujkhattar** [comment]
@ShrillShrestha Are you still planning to work on this?

**Comment by ShrillShrestha** [comment]
@tanujkhattar, I added the lines, and the tests are passing locally. Can you review it?

File: cirq-core/cirq/experiments/xeb_sampling_test.py
Comment (tanujkhattar) [inline cirq-core/cirq/experiments/xeb_sampling_test.py:37]:
```suggestion
            two_qubit_op_factory=lambda a, b, _: cirq.SQRT_ISWAP(a, b),
```

File: cirq-core/cirq/experiments/xeb_sampling_test.py
Comment (tanujkhattar) [inline cirq-core/cirq/experiments/xeb_sampling_test.py:24]: nit: note needed. You can simply do `cirq.SQRT_ISWAP`

File: cirq-core/cirq/experiments/xeb_simulation_test.py
Comment (tanujkhattar) [inline cirq-core/cirq/experiments/xeb_simulation_test.py:24]: same

File: cirq-google/cirq_google/calibration/xeb_wrapper_test.py
Comment (tanujkhattar) [inline cirq-google/cirq_google/calibration/xeb_wrapper_test.py:123]:
```suggestion
            two_qubit_op_factory=lambda a, b, _: (cirq.SQRT_ISWAP_INV.on(a, b)),
```

File: cirq-google/cirq_google/optimizers/convert_to_sqrt_iswap.py
Comment (tanujkhattar) [inline cirq-google/cirq_google/optimizers/convert_to_sqrt_iswap.py:206]:
```suggestion
    yield cirq.SQRT_ISWAP_INV(a, b)
```

File: dev_tools/notebooks/isolated_notebook_test.py
Comment (tanujkhattar) [inline dev_tools/notebooks/isolated_notebook_test.py:46]: I think this is actually not needed anymore because `cirq.SQRT_ISWAP` and `cirq.SQRT_ISWAP_INV` changes have already been [released](https://github.com/quantumlib/Cirq/releases/tag/v0.12.0). You can remove these and also revert the `pip install --pre` changes. 

This PR has been stalling for a while :)

**Review by tanujkhattar — CHANGES_REQUESTED** [review]
1. Use `cirq.SQRT_ISWAP_INV` instead of `cirq.SQRT_ISWAP ** -1` everywhere in the code.
2. `cirq.SQRT_ISWAP` was part of the previous release and hence the notebooks do not depend on any unreleased features anymore. Let's revert the two changes -- `pip install --pre` and the note after notebook setup.

File: cirq-core/cirq/experiments/xeb_sampling_test.py
Comment (ShrillShrestha) [reply cirq-core/cirq/experiments/xeb_sampling_test.py:37]: @tanujkhattar, sorry to bother you again.  But I have some questions.
As per the https://github.com/quantumlib/Cirq/issues/4223#issuecomment-863703185 it was mentioned:
> Within cirq-core, use ops.SQRT_ISWAP and use cirq.SQRT_ISWAP elsewhere (cirq.ops is used to avoid circular dependencies within cirq-core)
So, won't this create a circular dependency?  I am not sure if the latest changes have something to do with this.

File: cirq-core/cirq/experiments/xeb_fitting.py
Comment (ShrillShrestha) [inline cirq-core/cirq/experiments/xeb_fitting.py:167]: @tanujkhattar , Should I change these to `cirq.SQRT_ISWAP_INV` or `ops.SQRT_ISWAP_INV` as well?

File: cirq-core/cirq/experiments/xeb_sampling_test.py
Comment (tanujkhattar) [reply cirq-core/cirq/experiments/xeb_sampling_test.py:37]: Use `cirq.SQRT_ISWAP` in tests, because tests are not imported anywhere else and hence we directly `import cirq` in tests. For other files in cirq-core, import ops and use `ops. SQRT_ISWAP` (for files within the ops/* directy you'd directly have to import the file instead of importing module ops).

File: dev_tools/notebooks/isolated_notebook_test.py
Comment (ShrillShrestha) [reply dev_tools/notebooks/isolated_notebook_test.py]: I made the suggested changes, and all the test are passing locally. Thank you for helping me out.

File: cirq-core/cirq/experiments/xeb_fitting_test.py
Comment (tanujkhattar) [inline cirq-core/cirq/experiments/xeb_fitting_test.py:24]: not needed

File: cirq-core/cirq/experiments/xeb_fitting_test.py
Comment (tanujkhattar) [inline cirq-core/cirq/experiments/xeb_fitting_test.py:45]:
```suggestion
            q0, q1, depth=50, two_qubit_op_factory=lambda a, b, _: cirq.SQRT_ISWAP(a, b), seed=52
```

File: cirq-core/cirq/experiments/xeb_fitting_test.py
Comment (tanujkhattar) [inline cirq-core/cirq/experiments/xeb_fitting_test.py:170]:
```suggestion
@pytest.mark.parametrize('gate', [cirq.SQRT_ISWAP, cirq.FSimGate(np.pi / 4, 0)])
```

File: cirq-core/cirq/experiments/xeb_fitting_test.py
Comment (tanujkhattar) [inline cirq-core/cirq/experiments/xeb_fitting_test.py:218]:
```suggestion
            two_qubit_op_factory=lambda a, b, _: cirq.SQRT_ISWAP(a, b),
```

File: dev_tools/notebooks/isolated_notebook_test.py
Comment (tanujkhattar) [inline dev_tools/notebooks/isolated_notebook_test.py:43]: revert this change, it's not related to the PR.

**Review by tanujkhattar — CHANGES_REQUESTED** [review]
A few places were still missed. This should be good to go once this is fixed.

**Comment by ShrillShrestha** [comment]
@tanujkhattar, I made the changes. Hopefully, I did everything right this time. ;)

File: cirq-core/cirq/experiments/xeb_fitting.py
Comment (tanujkhattar) [reply cirq-core/cirq/experiments/xeb_fitting.py:167]: Yes, this should also change to `ops.SQRT_ISWAP_INV`.

**Review by tanujkhattar — APPROVED** [review]
LGTM % one final nit. Please fix it and we can merge.

**Pull request merged**
